src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +529,336 @@ private static String trimExcessWhitespaces(String text) {
+                for (DetailNode nodeChild : node.getChildren()) {
+                    if (nodeChild.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                        tagNode = nodeChild;
+                    }
+                }
+            }
+        }
+        return tagNode;
+    }
+
+    /**
+     * Checks if period is at the end of sentence.
+     *
+     * @param ast Javadoc root node.
+     * @return violation string
+     */
+    private static String getInlineSummarySentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        if (getDescriptionNode(ast) != null) {
+            for (DetailNode child : getDescriptionNode(ast).getChildren()) {
+                if (child.getType() == JavadocTokenTypes.TEXT) {
+                    result.append(child.getText());
+                }
+            }
+        }
+
+        if (getInlineTagNodeInsideHtml(ast) != null) {
+            final DetailNode descriptionNode = getDescriptionNodeInsideHtml(
+                getInlineTagNodeInsideHtml(ast));
+            if (descriptionNode != null) {
+                for (DetailNode child : descriptionNode.getChildren()) {
+                    if (child.getType() == JavadocTokenTypes.TEXT) {
+                        result.append(child.getText());
+                    }
+                }
+            }
+        }
+        return result.toString().trim();
+    }
+
+    /**
+     * Get Description node from ast.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode descriptionNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                for (DetailNode dataType : node.getChildren()) {
+                    if (dataType.getType() == JavadocTokenTypes.DESCRIPTION) {
+                        descriptionNode = dataType;
+                    }
+                }
+            }
+        }
+        return descriptionNode;
+    }
+
+    /**
+     * Finds and returns first sentence.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineFirstSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                result.append(returnInlineFirstSentence(node));
+            }
+
+            if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                result.append(returnInlineFirstSentence(
+                        Objects.requireNonNull(getInlineTagNodeInsideHtml(ast))));
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Sets result string in method getInlineFirstSentence.
```

Comment:
it doesnot set anythig, it is static method that does not side effect of Check, it just return value, it does not know in context of what it executing.

Reply:
Done